### PR TITLE
[Draft] Halborn audit fixes & deployment scripts

### PR DESCRIPTION
**Pull request by alienHunterOnMars**
- this is a work in progress

### CODE REVIEW

**Review by ShadoySV — APPROVED** [review]

**Review by ShadoySV — APPROVED** [review]

**Review by wedancedalot — DISMISSED** [review]

File: contracts/airdrop/src/contract.rs [hunk 1]
```diff
@@ -301,11 +348,6 @@ pub fn handle_delegate_astro_to_bootstrap_auction(
     let mut state = STATE.load(deps.storage)?;
     let mut user_info = USERS.load(deps.storage, &info.sender)?;
 
-    // CHECK :: HAS USER ALREADY WITHDRAWN THEIR REWARDS ?
```
Comment (wedancedalot) [inline contracts/airdrop/src/contract.rs:304]: Do we allow users to withdraw the rewards as many time as needed?

File: contracts/lockdrop/src/contract.rs [hunk 2]
```diff
@@ -1786,9 +1805,7 @@ fn update_user_lockup_positions_and_calc_rewards(
 
         let lockup_astro_rewards: Uint128;
```
Comment (ShadoySV) [inline contracts/lockdrop/src/contract.rs:1806]: Maybe drop it variable at all?

**Comment by wedancedalot** [comment]
TS deployment scripts look a bit messy, maybe it's a good idea to clean up a bit so we don't mix up any parameters during deployment to mainnet

File: contracts/airdrop/src/contract.rs (hunk 1 above)
Comment (alienHunterOnMars) [reply contracts/airdrop/src/contract.rs:304]: not needed imo, this airdrop is one-time only, so better to have user call the `WithdrawAirdropReward` / `Claim` (after auction) fn once and we send him all his withdrawable ASTRO airdrop tokens

File: contracts/lockdrop/src/contract.rs (hunk 2 above)
Comment (alienHunterOnMars) [reply contracts/lockdrop/src/contract.rs:1806]: done

**Comment by alienHunterOnMars** [comment]
> TS deployment scripts look a bit messy, maybe it's a good idea to clean up a bit so we don't mix up any parameters during deployment to mainnet

yes i agree, not in their final state yet, will clean them up

**Pull request merged**